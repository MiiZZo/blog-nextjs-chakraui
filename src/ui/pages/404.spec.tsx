import React from "react";
import { cleanup, fireEvent } from "@testing-library/react";
import { myRender } from "@lib/testing/render";
import Page404 from "./404";

afterEach(cleanup);

describe("404 page", () => {
  it("should call the router back function", () => {
    const routerBack = jest.fn();

    const { getByTestId } = myRender(<Page404 />, { back: routerBack });
    const backButton = getByTestId("back");

    fireEvent.click(backButton);

    expect(routerBack).toBeCalled();
  });

  it("should match the component", () => {
    const { asFragment } = myRender(<Page404 />);

    expect(asFragment()).toMatchSnapshot();
  });
});
