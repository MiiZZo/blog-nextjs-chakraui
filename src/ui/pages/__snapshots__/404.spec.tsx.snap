// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 page should match the component 1`] = `
<DocumentFragment>
  <div
    class="css-3z75l1"
  >
    <h2
      class="css-p5x3r9"
    >
      404 | Page not found
    </h2>
    <div
      class="css-nmceeh"
    >
      <a
        href="/"
      >
        <button
          class="css-1vr2bmh"
          type="button"
        >
          Home
        </button>
      </a>
      <button
        class="css-jo9fku"
        data-testid="back"
        type="button"
      >
        Back
      </button>
    </div>
  </div>
</DocumentFragment>
`;
